cfn-resources/teams/cmd/resource/team-user/team_user.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teamuser
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231001001/admin"
+)
+
+func FilterOnlyValidUsernames(mongoDBCloudUsersAPIClient TeamUsersAPI, usernames []string) ([]atlasv2.CloudAppUser, *http.Response, error) {
```

Comment:
this is not filtering as much as it's validating as its currently used, it will fail at the first "not found user" and the whole function will error and the caller will exit with the error, can we either make sure this is the intended behavior and rename or fix the behavior to actually collect all valid names and operate over those?

Reply:
yes, I changed this on purpose because I believe this is the correct behaviour.
Now if the user inputs random text as `usernames` field, it will do best effort in finding usernames and then update the `teams` resource with whatever best effort users it found. This implies lots of "implicit unexpected" behaviour that customer may encounter if they actually accidentally made a typo in the username string..

so I agree we can change the name